<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8">

	<title>Maximum Likelihood Methods and Hypothesis Testing</title>
	<meta name="author" content="Filip Bielejec">

	<meta name="apple-mobile-web-app-capable" content="yes" />
	<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

	<link rel="stylesheet" href="../reveal/css/reveal.min.css">
	<link rel="stylesheet" href="../reveal/css/bootstrap.min.css">
	<link rel="stylesheet" href="../reveal/css/theme/filip.css" id="theme">
	<link rel="stylesheet" href="../reveal/css/font-awesome/css/font-awesome.min.css">
	<link rel="stylesheet" href="../reveal/lib/css/zenburn.css">

</head>

<body>

	<div style="position: absolute; top:10px; left:10px; z-index:100;">
		<a href="../index.html">
			<i class="fa fa-times-circle" style="color: #bbb;"></i>
		</a>
	</div>

	<div class="reveal">
		<div class="slides">

			<section data-background="#cce8cf" data-background-transition="none">
				<h2>Maximum Likelihood Methods and Hypothesis Testing</h2>
				<h3>in Molecular Phylogenetics</h3>
				<br>
				<p><a href="https://twitter.com/fbielejec">Filip Bielejec</a> / <a href="https://github.com/fbielejec">@fbielejec</a></p>
				<p><a href="http://rega.kuleuven.be/cev/ecv">ECV - Evolutionary and Computational Virology</a></p>
				<p><a href="https://www.kuleuven.be/english">KU Leuven</a></p>
			</section>

			<section data-background="#cce8cf" data-background-transition="none" data-transition="slide">
				<h3>Cuban Bioinformatics Workshop on</h3>
        <h4>Virus Evolution and Molecular Epidemiology</h4>
				<br>
				<span> Slides for the lectures availiable at:</span>
				<br>
				<a href="https://github.com/fbielejec/veme2016"> https://github.com/fbielejec/veme2016</a>
			</section>

		<section data-background="#cce8cf" data-background-transition="none" data-transition="none">
				<h2>Phylogenetics</h2>
			</section>

<!-- tree  -->

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Observed data</h3>
	<img class="stretch" src="images/observed.png">
</section>

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
		 <!-- directed acyclic graph -->
	 <!-- tree is a hypothesis based on observed data -->
	<h3>Phylogenetic tree</h3>
	<img class="stretch" src="images/tree1.png">
</section>

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<!-- series of mutation events leads to the observed data -->
	<h3>Mutation events generate observed data</h3>
	<img class="stretch" src="images/tree2.png">
</section>


<!-- substitution model -->

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Substitution models</h3>
<div class="row">
	<div class="col-sm-6">
		<a class="stretch"><img src="images/substitution.png"></a>
	</div>
	<div class="col-sm-6">
		<ul>
			<li>Continuous Time Markov Chain (CTMC) models describe character substitutions at particular site in the alignment.</li>
			<li class="fragment">Characterized by $K \times K$ rate matrix $\mathbf{Q}$.</li>
			<li class="fragment">Finite-time transition probability matrix $\mathbf{P}$ is obtained via matrix exponentiation: $P(t)=e^{Qt}=\underset{n=0}{\overset{\infty}{\sum}}Q^{n}\frac{t^{n}}{n!}$.</li>
		</ul>
	</div>
</div>
</section>

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Fabricating evolution</h3>
	<img class="stretch" src="images/simulating.png">
</section>


<!-- likelihood concept-->
<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Maximum likelihood inference</h3>
	<br>
			<p>
				<!-- theta represents some hypothesis -->
				<!-- if we considered the data fixed and parameter to be the variable -->
				Some data $D$ was generated by a model with a parameter $\theta$.
        The conditional probability of observing the data $\mathrm{Pr}(D \, | \, \theta)$, considered as a function of $\theta$
is called the likelihood function and denoted $\mathrm{L}(\theta \, | \, D)$.
</p>

<p class="fragment">
	To maximize the likelihood means to find a best parameter $\hat{\theta}$ which maximizes $\mathrm{L}(\theta \, | \, D)$.
	Such a parameter is then called the Maximum Likelihood Estimate (MLE).
</p>
</section>

<!-- coin tosses -->
<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Maximum likelihood inference</h3>
	<br>
			<p>
Let us consider a coin with head and tail, which is flipped $10$ times, and lands on it's head $8$ times. How to model the probability of such an outcome?
</p>

<p class="fragment">
	Data: $D=\left(k=8,n=10\right)$.
</p>

<p class="fragment">
	Parameter: probability $\theta$ of landing a head.
</p>

<p class="fragment">
	Likelihood: given by the <a href="https://en.wikipedia.org/wiki/Binomial_distribution">Binomial distribution</a>:
	$$\mathrm{L}(\theta  \, | \, k,n) = \left(\begin{array}{c}
\begin{array}{c}
n\\
k
\end{array}\end{array}\right)
  \, \theta^k \, (1-\theta)^{n-k}.$$
</p>

</section>

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Maximum likelihood inference</h3>
	<br>
	<img class="stretch" src="images/likelihood1.png">
			<p>
				The likelihood curve
</p>
<!-- who know what will it be? -->
<p class="fragment">
	The MLE?
</p>
</section>

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Maximum likelihood inference</h3>
	<br>
	<img class="stretch" src="images/likelihood2.png">
			<p>
				<!-- analytically by taking the derivative, setting it equal to 0, solving for parameter -->
				${\displaystyle \log L(\theta\,|\, k,n)=\log\binom{n}{k}+x\log(\theta)+(n-k)\log(1-\theta)}$
				${\displaystyle \frac{\partial\log L(\theta\,|\,8,10)}{\partial\theta}=\frac{8}{\theta}-\frac{10-8}{1-\theta}}=0\;\Longleftrightarrow\;\theta=0.8$
</p>
</section>

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Maximum likelihood inference</h3>
	<br>
	<!-- <img class="stretch" src="images/likelihood2.png"> -->
	<p>
		In phylogenetics, the data $D$ is the sequence alignment observed at the tips of the tree and the parameter $\Theta$ consists of
		 the substitution model parameters $\theta$,
		the phylogenetic tree topology $\tau$
		and its branch lengths $b$.
	</p>
</section>


<!-- Likelihood calculation on tree -->
<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Likelihood calculation on a fixed tree</h3>
	<div class="row">
		<div class="col-sm-6">
			<a class="stretch"><img src="images/pruning.png"></a>
		</div>
		<div class="col-sm-6">

			<p>
Assumptions: After divergence characters evolve independently, sites evolve independently.
			</p>

			<p class="fragment">
Inside brackets: probability of the data $TGAA$ observed at the tips and $x_{0}x_{1}x_{2}$ for the internal nodes.
			</p>

			<p class="fragment">
Outside brackets: sum over all possible path combinations in the state space $\mathcal{E}$.
			</p>

		</div>
	</div>
</section>

<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Likelihood calculation on a tree</h3>
	<div class="row">
		<div class="col-sm-6">
			<a class="stretch"><img src="images/pruning.png"></a>
		</div>
		<div class="col-sm-6">

			<p>
Ancestral reconstruction: Finding the most probable path between the unobserved conditional on the observed characters.
			</p>

				<p class="fragment">
This means maximizing the likelihood of the data: $P(TGAA | x_0,x_1,x_2)$.
<!-- $L({x_0,x_1,x_2)} | TGAA)$.  -->
			</p>

<!-- you would have to enumerate all and find the one that maximizes the probability -->
			<p class="fragment">
Computationaly intensive! Proceeds at $\mathcal{O}\left(K^{n-1}\right)$, where $K$ is the size of state space $\mathcal{E}$, $n$ is the number of internal nodes.
		</p>

		</div>
	</div>
</section>


<!-- tree pruning -->
<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Likelihood calculation on a tree</h3>
	<div class="row">
		<div class="col-sm-6">
			<a class="stretch"><img src="images/pruning2.png"></a>
		</div>
		<div class="col-sm-6">

			<p>
				Tree pruning [Felsenstein 1981]: recursive algorithm for calculating likelihood of a (observed) sequence on a tree.
			</p>

				<p class="fragment">
Partial likelihood $L_{i}(x_{i})$ of observing data at the descendants of node $i$ given state $x_{i}$ at node $i$ expressed in terms of partial likelihoods at nodes $j$ and $k$.
			</p>

<!-- from polynomial to quadratic -->
			<p class="fragment">
				Proceeds at $\mathcal{O}\left(K^{2} \times n \right)$.
		</p>

		</div>
	</div>
</section>

<!-- searching tree space -->
<section data-background="#cce8cf" data-background-transition="none" data-transition="none">
	<!-- by tree we understand topology and branch lengths -->
	<h3>What if the tree is not known?</h3>
</section>

<section data-background="none" data-background-transition="none" data-transition="none">
	<h3>Searching through tree space</h3>
	<div class="row">
		<div class="col-sm-6">
<a class="stretch"><img src="images/table1.png"></a>
		</div>
		<div class="col-sm-6">

			<p>
				Finding the tree that maximizes the likelihood of observing tip sequence requires a search through the space of all possible tree topologies and branch lengths.
			</p>

				<!-- <p class="fragment">
Impossible, but in the smallest cases
			</p> -->

		</div>
	</div>
</section>



<!-- newton rhapson for branch lengths -->
<section data-background="none" data-background-transition="none" data-transition="none">
	<h3>Searching through tree space</h3>
   <br>
			<p>
				Branch lengths $b$ can computed numerically by maximizing the log-likelihood of the whole alignment $X$,
				which is a sum of the log-likelihoods at the particular sites:
        $l(b | X,\tau)=\underset{j=1}{\overset{l}{\sum}}log\left(L(b | \mathbf{x}_{j},\tau)\right)$
			</p>

				<p class="fragment">
					This means finding those branch lengths for tree $\tau$ which maximize the log-likelihood of observing the data $X$.
			  </p>

		 	<p class="fragment">
        Accomplished by numerical routines like Newton-Raphson.
	  	</p>

</section>



<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Newton-Raphson (one dimension)</h3>
   <br>
			<p>
				Let's apply the Newton Raphson method to find the parameter $\theta$ which maximizes the function:
				$f(\theta)=\frac{e^{\theta}}{(1+e^{\theta})^{2}}$
			</p>

<p class="fragment">
Method starts with a function $f$, the function's derivative $f'$ (gradient), second order derivative $f''$ (hessian) and an initial guess
$\theta_{0}$ for the optimum of $f$.
</p>

<p class="fragment">
	At every iteration:
$\theta_{n+1}=\theta_{n}-\frac{f'\left(\theta_{n}\right)}{f''\left(\theta_{n}\right)}$
until some convergence criteria is reached.
 </p>

			<p class="fragment">
<font color="#df8f09">Let's see it in action!</font>
		</p>

</section>


<!-- summarize heuristic algs p 190 from the the book -->
<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Searching through tree space</h3>
   <br>
			<p>
				Branch lengths can be computed numerically by maximizing the log-likelihood for a single tree.
				The big task that remains is to actually find the tree among all possible tree topologies that maximizes the global likelihood.
			</p>

			<p class="fragment">
				No efficient algorithms exist that guarantee to find the ML tree, even for moderately sized data sets.
		 </p>

		<p class="fragment">
		   Most programs resort to various heuristic approaches to suggest reasonable trees.
    </p>

</section>


<section data-background="#cce8cf" data-background-transition="none" data-transition="none">
	<h3>Heuristic searches of the tree space</h3>
</section>


<section data-background="none" data-background-transition="none" data-transition="none">
	<h3>Stepwise addition search</h3>
	<div class="row">
		<div class="col-sm-6">
<a class="stretch"><img src="images/stepwise_addition.png"></a>
		</div>
		<div class="col-sm-6">

<!-- procedure starts from -->
			<p>
				Start from unrooted tree of 3 taxa, randomly selected from $n$, reconstruct corresponding ML sub-tree.
			</p>

			<p class="fragment">
				Randomly pick one of the remaining $n-3$ taxa, insert into each branch of the ML sub-tree.
		 </p>

		 <p class="fragment">
			 From the list of $2k-n$ trees, where $k$ is the number of added taxa, select the one which maximizes the likelihood.
		</p>

		</div>
	</div>
</section>


<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Stepwise addition search</h3>
	<div class="row">
		<div class="col-sm-6">
<a class="stretch"><img src="images/stepwise_addition.png"></a>
		</div>
		<div class="col-sm-6">

			<p>
				After $n-3$ steps a maximum likelihood tree is obtained, which is at least a local optima.
			</p>

			<!-- it is possible that another insertion order of the taxa will provide a tree with a higher likelihood. -->
			<p class="fragment">
				The result is highly dependent on the insertion order.
		 </p>

		 <p class="fragment">
			To reduce the risk of getting stuck in local maximum tree-rearrangement operations acting on the full tree were suggested.
		</p>

		</div>
	</div>
</section>


<!-- style="width: 100%; height: 100%;" -->

<!-- see p283 the book -->
<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Full tree rearrangement </h3>
<a class="stretch"><img src="images/rearrangement.png"></a>
			<p>
				Generate a number of trees from a starting tree (the neighborhood of the starting tree) using three operations:
			</p>
</section>


<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Full tree rearrangement </h3>
<a class="stretch"><img src="images/rearrangement.png"></a>
			<p>
		<font color="#df8f09">	Nearest neighbor interchange (NNI):</font>
			Visits each branch, swapping a subtree connected to one end of the branch with a subtree connected to
			the other end of the branch, resulting in two different topologies, compares th elikelihood to the original tree.
			</p>
</section>


<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Full tree rearrangement </h3>
<a class="stretch"><img src="images/rearrangement.png"></a>
			<p>
				<font color="#df8f09">	Subtree pruning and regrafting (SPR):</font>
				Involves clipping off all possible subtrees
			 from the main tree and reinserting them at all possible locations, avoiding repetitions.
			</p>
</section>


<section data-background="images/overImg.png" data-background-transition="none" data-transition="none">
	<h3>Full tree rearrangement </h3>
<a class="stretch"><img src="images/rearrangement.png"></a>
			<p>
				<font  size = 5.5 color="#df8f09">	Tree bisection and reconnection (TBR):</font>
				<font size = 5.5>Involves cutting a tree into two subtrees by cutting one branch, and reconnecting the two
subtrees by creating a new branch that joins a branch on one subtree to a branch
on the other sub-tree.
 All possible branches are tried, avoiding repetitions.
 </font>
			</p>
</section>


<section data-background="#cce8cf" data-background-transition="none" data-transition="none">
	<h3>Assessing evolutionary trees and models </h3>
</section>


<section data-background="none" data-background-transition="none" data-transition="none">
	<h3>Are two evolutionary trees/models different?</h3>
<p>
We already know how given a sequence alignment data and a substitution models to reconstruct
trees and compute their likelihoods.
</p>
<br>
<p class="fragment">
	But can we decide from the likelihood:
	<ul class="fragment">
		<li>Which substitution model better fits the data?</li>
		<li> Which reconstructed tree is better (in terms of their likelihoods)?</li>
		<li> Whether one tree likelihood is significantly better/worse?</li>
	</ul>
</p>
<br>
<p class="fragment">
	These questions can be assesed by performing hypothesis tests.
</p>
</section>


<section data-background="none" data-background-transition="none" data-transition="none">
<h3>Likelihood ratio test (LRT)</h3>
<div class="row">
	<div class="col-sm-6">
<a class="stretch"><img src="images/lrt.png"></a>
	</div>
	<div class="col-sm-6">
		<p>
			<font size=4>
			If two models are nested (like TN in GTR), their log-likelihood difference $2\left(l\left(\text{GTR}\right)-l\left(\text{TN}\right)\right)$
		follows a $\chi^{2}$ distribution with degrees of freedom equal to the number of additional
		parameters ($df = 3$).
	</font>
		</p>

		<p>
			<font size=4>
				If the observed value falls into
				the extreme 5% interval (shaded
				area), we say the more complex
				model (GTR) is significatly
				better than the simpler (TN).
		</font>
		</p>

		<p>
			<font size=4>
				<font>$H_{0}$</font>: both models are equally good.
				<br>
				<font>$H_{A}$</font>: the more complex model is better.
		</font>
		</p>

	</div>
</div>
</section>


<section data-background="none" data-background-transition="none" data-transition="none">
	<h3>Testing tree topologies</h3>

<!-- likelihood diff had unknown distribution -->
<p>
	Two different topologies are not nested, thus, we cannot use the$\chi^{2}$ distribution.
</p>

<p class="fragment">
	Hence, bootstrap-based methods have been applied to
determine the distribution of log-likelihood differences for testing.
</p>

</section>

<!--  bootstrap -->
<!-- bootstrap because of baron muchausen -->
<section data-background="none" data-background-transition="none" data-transition="none">
	<h3>Bootstrap - basic idea</h3>

	<ul>
		<li> Compute log-likelihood values $l_{1},\ldots,l_{N}$ for tree topologies $\tau_{1},\ldots,\tau_{N}$.</li>
		<li>
			Draw 	<font color="#df8f09">bootstrap samples</font> $i$ from the alignment,
			re-estimate the log-likelihood values $l_{X}^{\left(i\right)}$
      for each tree $T_{x}$ and for each sample $i$.
		</li>

		<li>
Center the log-likelihoods with the mean $\overline{l}_{X}^{\left(i\right)}$ by setting
$\tilde{l}_{X}^{\left(i\right)}=l_{X}^{\left(i\right)}-\overline{l}_{X}^{\left(i\right)}$.
		</li>
<li>
		Use the differences between the $\tilde{l}_{X}^{\left(i\right)}$ to determine the <font color="#df8f09">non-parametric distribution</font> of differences:
		$\delta^{\left(i\right)}=\tilde{l}_{Y}^{\left(i\right)}-\tilde{l}_{Z}^{\left(i\right)}$.
	</li>

	<li>
			Use the distribution of $\delta^{\left(i\right)}$ to test the trees.
		</li>

	</ul>

</section>


<!-- TODO: bootstrap example-->



			<section data-background="images/overImg.png" data-background-transition="none">
				<!-- very happy  to answer all questions you might have -->
				<h2>Thank you!</h2>
				<h2><font color="#df8f09"> Questions?</font></h2>
			</section>

		</div>
	</div>

	<script src="../reveal/lib/js/head.min.js"></script>
	<script src="../reveal/js/reveal.min.js"></script>
	<script src="../reveal/js/config.js"></script>

</body>

</html>
